src/main/java/io/confluent/connect/jdbc/dialect/DatabaseDialect.java

```diff
@@ -360,6 +360,21 @@ String buildUpsertQueryStatement(
       Collection<ColumnId> nonKeyColumns
   );
 
+  /**
+   * Build the DELETE prepared statement expression for the given table and its columns. Variables
+   * for each key column should also appear in the WHERE clause of the statement.
+   *
+   * @param table         the identifier of the table; may not be null
+   * @param keyColumns    the identifiers of the columns in the primary/unique key; may not be null
+   *                      but may be empty
+   * @return the delete statement; may not be null
+   * @throws UnsupportedOperationException if the dialect does not support upserts
+   */
+  String buildDeleteStatement(
+      TableId table,
+      Collection<ColumnId> keyColumns
+  );
```

Comment:
Should this be a default method that throws UnsupportedOperationException? It's possible that other projects depend on this connector and add their own dialect, and adding a non-default method would break those projects. Yeah, it's likely they extend `GenericDatabaseDialect`, but we can't rule it out. Plus, I think this makes sense, too, because by default dialects (that don't extend `GenericDatabaseDialect`) probably won't support deletes anyway.

Reply:
good idea - fixed